tests/test_staging/test_validation.py

```diff
@@ -73,47 +76,33 @@ def test_validate_post_col(test_data_df, monkeypatch, caplog):
         assert "Invalid pattern postcodes found: ['HIJ 789']" in caplog.text
 
     # Valid AND real postcodes
-    df_valid = pd.DataFrame({"referencepostcode": ["NP10 8XG", "PO15 5RR", "SW1P 4DF"]})
+    df_valid = pd.DataFrame(
+        {
+            "reference": [1, 2, 3],
+            "instance": [0, 0, 0],
+            "referencepostcode": ["NP10 8XG", "PO15 5RR", "SW1P 4DF"],
+        }
+    )
     assert validate_post_col(df_valid, fake_path)
 
     # Invalid postcodes
-    df_invalid = pd.DataFrame({"referencepostcode": ["EFG 456", "HIJ 789"]})
-    with pytest.raises(ValueError) as error:
-        validate_post_col(df_invalid, fake_path)
-    assert str(error.value) == "Invalid postcodes found: ['EFG 456', 'HIJ 789']"
+    df_invalid = pd.DataFrame(
+        {
+            "reference": [1, 2],
+            "instance": [0, 0],
+            "referencepostcode": ["EFG 456", "HIJ 789"],
+        }
+    )
+    validate_post_col(df_invalid, fake_path)
+    assert "Invalid postcodes found: ['EFG 456', 'HIJ 789']" in caplog.text
 
     # Mixed valid and invalid postcodes - as is in the test_data
-    with pytest.raises(ValueError) as error:
-        validate_post_col(test_data_df, fake_path)
+
+    validate_post_col(test_data_df, fake_path)
     if config["global"]["postcode_csv_check"]:
```

Comment:
Line 139: alternative name for unreal_postcodes: result_df
Line 156: Checking that none of the valid post codes are in unreal_postcodes - this is potentially a repetition of test_check_pcs_real_with_invalid_postcodes